pkg/cloudevents/transport/http/transport.go

```diff
@@ -599,18 +598,39 @@ func (t *Transport) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	r.OK()
 }
 
-// GetPort returns the port the transport is active on.
-// .Port can be set to 0, which means the transport selects a port, GetPort
-// allows the transport to report back the selected port.
+// GetPort returns the listening port.
+// Returns -1 if there is a listening error.
+// Note this will call net.Listen() if  the listener is not already started.
 func (t *Transport) GetPort() int {
-	if t.Port != nil && *t.Port == 0 && t.realPort != 0 {
-		return t.realPort
+	// Ensure we have a listener and therefore a port.
+	if t.listen() == nil || t.Port != nil {
+		return *t.Port
 	}
+	return -1
+}
 
-	if t.Port != nil && *t.Port >= 0 { // 0 means next open port
-		return *t.Port
+func (t *Transport) setPort(port int) {
+	if t.Port == nil {
+		t.Port = new(int)
 	}
-	return 8080 // default
+	*t.Port = port
+}
+
+// listen if not already listening, update t.Port
+func (t *Transport) listen() (err error) {
```

Comment:
this method should be:

```golang
func (t *Transport) listen() (string, error)
```

returning the addr and the error.

Reply:
done